PSICompile/ILCodeGen.cs

```diff
@@ -53,10 +67,15 @@ public class ILCodeGen : Visitor {
    }
 
    void StoreVar (Token name) {
-      var vd = (NVarDecl)mSymbols.Find (name)!;
-      var type = TMap[vd.Type];
-      if (vd.Local) Out ($"    stloc {vd.Name}");
-      else Out ($"    stsfld {type} Program::{vd.Name}");
+      switch (mSymbols.Find (name)) {
+         case NVarDecl v:
+            string type = TMap[v.Type];
+            if (v.Local) Out ($"    stloc {v.Name}");
+            else Out ($"    stsfld {type} Program::{v.Name}"); 
+            break;
+         case NFnDecl f: break;

```

Comment:
Test with this function:
```
function Min2 (a: integer, b: integer): integer; 
begin 
  if a < b then
    Min2 := a;
  else
    Min2 := b;
end;
```

Reply:
program Comp12;
var
  g: integer;

function Min2 (a, b: integer): integer;
begin
  if a < b then
    Min2 := a;
  else
    Min2 := b;
end;

begin
  g := Min2 (13, 5);
  WriteLn ("Minimum (13, 5) = ", g);
  if 3 < 4 then begin
     WriteLn ("Three is less than four.");
          WriteLn ("But you know that already.");
  end;
end.
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Minimum (13, 5) = 5
Three is less than four.
But you know that already.
────────────────────────────────
This works.

Reply:
OK. What about this valid Pascal program. I think it will result in an invalid IL code. 
```
function Min3 (a: integer; b: integer): integer;
begin
   Min3 := b;
   if a < b then Min3 := a;
end;
```

Reply:
Yes this throws exception if I call Min3 (5, 13).